torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1202,6 +1202,10 @@ struct to_ir {
         if (expr_out->node()->kind() == aten::is_scripting) {
           static_if = true;
         }
+        // MetaCompile on boolean literals and constants
```

Comment:
line169 has some comment that appears related and confusing:
```
  c10::optional<bool>
      static_if_; // certain expression cause us to emit a static if statement
                  // this value is present if this is the case.
                  // this is not equivalent to value_ being a constant
                  // it is possible for value_ to be constant but for
                  // the expression that produced it to not trigger the
                  // static if behavior. e.g. use of a variable assigned
                  // to a constant
```
I don't know if they need to be updated to reflect this PR